code/controllers/subsystem/ping.dm

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) 2022 Aleksej Komarov
+ * SPDX-License-Identifier: MIT
+ */
+
+SUBSYSTEM_DEF(ping)
+	name = "Ping"
+	priority = FIRE_PRIORITY_PING
+	wait = 4 SECONDS
+	flags = SS_NO_INIT
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	var/list/currentrun = list()
```

Comment:
I wonder if we should remove from this in client/Destroy, since clients instantly delete, it would hang a reference

Reply:
SS discards clients (including null clients) from this batch as it processes them until it becomes empty and repeats the cycle.

Reply:
yes, but when a client logs out, it is instantly hard deleted.
it's important that clients need to clear as little references as possible. You'll need to empty it out in client/Destroy